README.md

```diff
@@ -1,38 +1,137 @@
-eventtarget.js
-========
+# eventtarget
 
-#### JavaScript events for custom objects ####
+JavaScript events for custom objects
 
-### Usage ###
+## Installation
+
+### Browser
 
 ```html
 <script src="EventTarget.js"></script>
-<script>
+```
 
-	// Applying EventTarget to custom object
+### Node
 
-	var Car = function () {
+```shell
+npm install
+```
 
-		EventTarget.call( this );
+```js
+var EventTarget = require('eventtarget');
+```
 
-		this.start = function () {
+## Usage
 
-			this.dispatchEvent( { type: 'started', foo: 'bar' } );
+```html
+<script>
 
-		};
+// Adding events to custom object
 
-	};
+var Car = function () {
+  this.start = function () {
+    this.dispatchEvent({type: 'start', message: 'vroom vroom!'});
+  };
+};
 
+Object.assign(Car.prototype, EventTarget.prototype);
 
-	// Using events
+// Using events
 
-	var car = new Car();
-	car.addEventListener( 'started', function ( event ) {
+var car = new Car();
 
-		alert( 'vroom vroom!' );
+car.addEventListener('start', function (event) {
+  alert(event.message);
+});
 
-	} );
-	car.start();
+car.start();
 
 </script>
 ```
+
+## Constructor
+
+If you inherit from `EventTarget`, you can invoke the (non-standard)
+constructor with options. See `setOptions`.
+
+## Standard methods
+
+The following behave as with the standard `EventTarget` methods
+(as possible). Options supported include the standard options, `capture`
+(including expressed as a boolean), as well as `once` (which auto-removes
+the listener after execution) and `passive` (which prevents the listener
+from using `preventDefault` even if the event passed to the target was
+`cancelable`).
+
+-   `addEventListener(type, listener, options)` - These standard methods cannot
+    prevent execution of the non-standard
+    `addDefaultEventListener`/`addLateEventListener`-added methods.
+
+-   `removeEventListener(type, listener, options)`
+
+-   `dispatchEvent(ev)` - Will check as appropriate the event's `type`,
+    `bubbles` or `cancelable` (which can all be set with invocation of the
+    constructor, `Event`) or the readonly `eventPhase` and `defaultPrevented`.
+    The properties `target`, `currentTarget`, `eventPhase`, and
+    `defaultPrevented` will be set as appropriate.
+
+-   `on* = function (ev) {};` (e.g., `onclick`) - Can use `return false` to
+    prevent the default (but this will not stop propagation or stop immediate
+    propagation as per regular JavaScript behavior (unlike jQuery which stops
+    propagation)).
+
+## Custom methods
```

Comment:
Fair enough--most of the functionality I added could be implemented--at some inconvenience to users--by allowing the class to be created with the early, late, and default listeners and options baked in when (dynamically) creating the class. And the `WeakMap`/private properties would work fine.

The only thing which cannot be done fully by spec is to allow (as did my PR) passing in of native Event objects, as their properties must first be copied into a new object in order to allow setting of read-only `target`, etc. properties, but then the event which is passed on is not identical to the original object. Not accepting native events is non-standard behavior too though, so foregoing the copying would also be non-standard.

Anyways, I made a note as a possible to-do, but I have already spent too much time on this fork for now as it is and am happy that I can use it for my needs, so you can close this PR as you like.